src/components/Settings/ChatAppTab.react.js

```diff
@@ -28,10 +29,14 @@ class ChatAppTab extends React.Component {
   };
 
   handleSubmit = () => {
-    const { actions } = this.props;
+    const { actions, userEmailId } = this.props;
     const { enterAsSend } = this.state;
     this.setState({ loading: true });
-    setUserSettings({ enterAsSend })
+    let payload = {
+      enterAsSend,
+    };
+    payload = userEmailId !== '' ? { ...payload, email: userEmailId } : payload;
```

Comment:
I think we can always send the email in payload. We can have a check while setting in store, if the email is equal to logged in user email, then use the store that already existed, else use `userSettingsViewedByAdmin` object.